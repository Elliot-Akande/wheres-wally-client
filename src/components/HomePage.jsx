import { Link } from "react-router-dom";
import useLevelData from "../hooks/useLevelData";

function HomePage() {
  const { levelData, error } = useLevelData();

  return (
    <>
      {!error ? (
        levelData.map(({ levelNum }) => (
          <Link to={`level/${levelNum}/details`} key={levelNum}>
            Level {levelNum}
          </Link>
        ))
      ) : (
        <p>Data fetching error.</p>
      )}
    </>
  );
}

export default HomePage;
