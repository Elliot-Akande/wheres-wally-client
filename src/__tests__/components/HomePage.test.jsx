import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { vi } from "vitest";
import HomePage from "../../components/HomePage";

const mocks = vi.hoisted(() => {
  return {
    useLevelData: vi.fn(),
  };
});

vi.mock("../../hooks/useLevelData", () => {
  return {
    default: mocks.useLevelData,
  };
});

describe("HomePage", () => {
  it("renders Level links", () => {
    mocks.useLevelData.mockReturnValue({
      levelData: [
        {
          levelNum: 1,
        },
        {
          levelNum: 2,
        },
        {
          levelNum: 3,
        },
      ],
      error: null,
    });

    const { container } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders error message on useLevelData error", () => {
    mocks.useLevelData.mockReturnValue({
      levelData: [],
      error: new Error(),
    });

    const { container } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
