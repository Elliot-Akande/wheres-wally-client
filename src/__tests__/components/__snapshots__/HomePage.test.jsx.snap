// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > renders Level links 1`] = `
<div>
  <a
    href="/level/1/details"
  >
    Level 
    1
  </a>
  <a
    href="/level/2/details"
  >
    Level 
    2
  </a>
  <a
    href="/level/3/details"
  >
    Level 
    3
  </a>
</div>
`;

exports[`HomePage > renders error message on useLevelData error 1`] = `
<div>
  <p>
    Data fetching error.
  </p>
</div>
`;
